python/cudf/cudf/utils/utils.py

```diff
@@ -225,6 +101,10 @@ def __array_function__(self, *args, **kwargs):
         return False
 
 
+# TODO: Do we need to expose this? It's essentially a trivial wrapper for an
+# rmm function and we don't use it anywhere internally. Unless this is intended
+# to allow us the flexibility to use allocators other than RMM in the future it
+# doesn't serve much purpose.
```

Comment:
This is part of the public API is the reason this exists. It's exposed under `cudf.set_allocator`.

Reply:
Right, my question is why it needs to be. Do we want the flexibility to support other allocators in the future, or at least to have conceptual abstraction between `cudf` and `rmm` so that users don't have to think about `rmm` at all if they don't want to? Otherwise we could just have users call `rmm.set_allocator` themselves.

Reply:
I think @shwina introduced this and had reasons? I don't quite remember the reasoning as to why here.

Reply:
I think the reasoning was that we didn't want to introduce yet another concept/library for users to understand.

Reply:
Comment still needs to be removed before we resolve this.

Reply:
My mistake, I decided to run tests locally before pushing for once then promptly forgot to push.